### PR TITLE
feat(macos): Build for macOS [LOG-13025]

### PR DESCRIPTION
**Pull request by james-t-bloom**
feat(macos): Build for macOS

Summary: Ended up being mostly cargo make, some cargo.toml changes and
a lot of test ignores
Makes it so agent builds on Mac with a test suite

Test plan:

Manually testing
unit tests
Int tests

Ref: LOG-13025

### CODE REVIEW

**Comment by dkhokhlov** [comment]
very big PR - is it possible to split it?

**Comment by james-t-bloom** [comment]
> very big PR - is it possible to split it?

Not really - but it's not as big as it seems. Mostly ignoring tests with cfg - and the ignoring has a cascading effect for lint reasons.

File: common/fs/src/cache/mod.rs [hunk 1]
```diff
@@ -1896,7 +1901,7 @@ mod tests {
 
     // Deletes a directory
     #[tokio::test]
-    #[cfg(unix)]
+    #[cfg(target_os = "linux")]
     async fn filesystem_delete_filled_dir() -> io::Result<()> {
         let tempdir = TempDir::new()?;
```
Comment (dkhokhlov) [inline common/fs/src/cache/mod.rs:1906]: why disabled on mac?

File: common/notify_stream/src/lib.rs [hunk 2]
```diff
@@ -191,6 +191,9 @@ impl Watcher {
                     EventKind::Modify(ModifyKind::Data(DataChange::Any)) => {
                         Some(Event::Write(received.paths.first().unwrap().clone()))
                     }
+                    EventKind::Modify(ModifyKind::Data(DataChange::Content)) => {
+                        Some(Event::Write(received.paths.first().unwrap().clone()))
```
Comment (dkhokhlov) [inline common/notify_stream/src/lib.rs:195]: did we miss it from the beginning?

File: common/notify_stream/src/lib.rs [hunk 3]
```diff
@@ -586,50 +593,6 @@ mod tests {
         Ok(())
     }
 
-    /// macOS will follow symlink files
-    #[tokio::test]
-    #[cfg(target_os = "macos")]
-    async fn test_watch_symlink_write_after_create_macos() -> io::Result<()> {
```
Comment (dkhokhlov) [inline common/notify_stream/src/lib.rs:592]: removed for mac?

File: bin/src/main.rs [hunk 4]
```diff
@@ -1,13 +1,9 @@
...
 use std::sync::Arc;
 use tokio::sync::Mutex;
-
-#[cfg(unix)]
-use tracing::{debug, info, trace, warn};
-
-#[cfg(windows)]
 use tracing::info;
-
+#[cfg(any(target_os = "windows", target_os = "linux"))]
```
Comment (dkhokhlov) [inline bin/src/main.rs:5]: cfg(not(target_os = "macos")) ?

File: common/notify_stream/src/lib.rs (hunk 2 above)
Comment (james-t-bloom) [reply common/notify_stream/src/lib.rs:195]: no - as far as I can tell it's a mac thing.

File: common/fs/src/cache/mod.rs (hunk 1 above)
Comment (james-t-bloom) [reply common/fs/src/cache/mod.rs:1902]: There was kind of a cut bait and run strategy on some of these tests - Timing and directory creation issues that were found to be on the test side not code side.

File: common/notify_stream/src/lib.rs (hunk 3 above)
Comment (james-t-bloom) [reply common/notify_stream/src/lib.rs:592]: I don't know where this test came from - but it didn't work on macos

Comment (dkhokhlov) [reply common/notify_stream/src/lib.rs:592]: Lets track all Mac problems in JIRA (create separate tickets). Disabled tests, symptoms, err cause if any.

Comment (james-t-bloom) [reply common/notify_stream/src/lib.rs:592]: yup will do

File: bin/tests/it/cmd_line_env.rs [hunk 5]
```diff
@@ -10,6 +10,8 @@ use std::path::Path;
 use std::process::{Command, Output, Stdio};
 use std::str::from_utf8;
 use std::sync::{Arc, Condvar, Mutex};
+#[cfg(any(target_os = "windows", target_os = "linux"))]
```
Comment (stsantilena) [inline bin/tests/it/cmd_line_env.rs:13]: Is there a "not mac" option?

File: Makefile.toml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+[env]
```
Comment (stsantilena) [inline Makefile.toml:1]: Can you explain a bit why the need for the this new `Makefile.toml` file? How is it different from a traditional Makefile?

**Review by stsantilena — APPROVED** [review]
Just a couple small questions. Other than that, LGTM.

File: bin/tests/it/cmd_line_env.rs (hunk 5 above)
Comment (james-t-bloom) [reply bin/tests/it/cmd_line_env.rs:13]: I honestly just did what the linter suggested, but probably would have been the way to go

File: Makefile.toml (hunk 6 above)
Comment (james-t-bloom) [reply Makefile.toml:1]: so its a part of this cargo make plug in, it kind of augments the make file. Use it mostly for convenience cargo make run, cargo make int-tests etc

**Pull request merged**